src/applications/gi/actions/index.js

```diff
@@ -167,7 +168,32 @@ export function fetchSearchResults(query = {}) {
     return fetch(url, api.settings)
       .then(res => res.json())
       .then(
-        payload => withPreview(dispatch, { type: SEARCH_SUCCEEDED, payload }),
+        payload =>
+          withPreview(dispatch, {
+            type: INSTITUTION_SEARCH_SUCCEEDED,
+            payload,
+          }),
+        err => dispatch({ type: SEARCH_FAILED, err }),
+      );
+  };
+}
+
+export function fetchProgramSearchResults(query = {}) {
+  const queryString = Object.keys(query).reduce((str, key) => {
+    const sep = str ? '&' : '';
+    return `${str}${sep}${snakeCase(key)}=${query[key]}`;
+  }, '');
+
+  const url = `${api.url}/institution_programs/search?${queryString}`;
```

Comment:
Would importing `appendQuery` from the `append-query` module serve your needs here?

Reply:
Yes, thanks!